src/watchmaker/workers/yum.py

```diff
@@ -54,40 +54,15 @@ def _get_amazon_el_version(self, version):
 
     def get_dist_info(self):
         """Validate the Linux distro and return info about the distribution."""
-        dist = None
-        version = None
+        dist = distro.id()
+        version = distro.version()[0]
         el_version = None
 
-        # Read first line from /etc/system-release
-        try:
-            with io.open('/etc/system-release', encoding='utf8') as fh_:
-                release = fh_.readline().strip()
-        except Exception:
-            self.log.critical(
-                'Failed to read /etc/system-release. Cannot determine system '
-                'distribution!'
-            )
-            raise
-
-        # Search the release file for a match against _supported_dists
-        matched = self.DIST_PATTERN.search(release.lower())
-        if matched is None:
-            # Release not supported, exit with error
-            msg = (
-                'Unsupported OS distribution. OS must be one of: {0}'
-                .format(', '.join(self.SUPPORTED_DISTS))
-            )
-            self.log.critical(msg)
-            raise WatchmakerError(msg)
```

Comment:
We should still raise and exit with error if `dist` is not one of the `SUPPORTED_DISTS`.

Reply:
Clapper's whacking my test-EC2s. This _should_ satisfy the requirement:

~~~
       # Error if 'dist' does not found in SUPPORTED_DISTS
       if dist not in SUPPORTED_DISTS
           # Release not supported, exit with error
           msg = (
               'Unsupported OS distribution. OS must be one of: {0}'
               .format(', '.join(self.SUPPORTED_DISTS))
           )
           self.log.critical(msg)
           raise WatchmakerError(msg)
~~~

?

At any rate, once it's after 1830, I'll do test-launches to validate.

Reply:
That looks right, yes

Reply:
Alright, bout to launch new EC2s to see if this makes anything vomit in either RHEL 8 or CentOS 8-stream.

Reply:
Ok, so `SUPPORTED_DISTS` is defined in the `Yum` class but it's not available to the `get_dist_info` function in that class?

Reply:
Oh... `self.SUPPORTED_DISTS`?

Reply:
Alright, working nao.